packages/test/test-end-to-end-tests/src/test/deltaManagerProxy.spec.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import { SharedString, Side } from "@fluidframework/sequence/internal";
+import {
+	TestFluidObjectFactory,
+	type ITestFluidObject,
+	type ITestObjectProvider,
+} from "@fluidframework/test-utils/internal";
+import type { IContainerRuntimeOptions } from "@fluidframework/container-runtime/internal";
+import type { ILoaderProps } from "@fluidframework/container-loader/internal";
+
+describeCompat("Container", "NoCompat", (getTestObjectProvider, apis) => {
+	const { ContainerRuntimeFactoryWithDefaultDataStore } = apis.containerRuntime;
+	const loaderProps: Partial<ILoaderProps> = {
+		options: {
+			intervalStickinessEnabled: true,
+		},
+	} as unknown as ILoaderProps;
+
+	const sharedType = "sharedString";
+	let provider: ITestObjectProvider;
+	const defaultFactory = new TestFluidObjectFactory([[sharedType, SharedString.getFactory()]]);
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
+	};
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
+		defaultFactory,
+		registryEntries: [[defaultFactory.type, Promise.resolve(defaultFactory)]],
+		runtimeOptions,
+	});
+
+	const assertConsistent = (...sharedStrings: SharedString[]) => {
+		const text = sharedStrings[0].getText();
+		for (let i = 1; i < sharedStrings.length; i++) {
+			assert.equal(sharedStrings[i].getText(), text, `SharedString ${i} is inconsistent`);
+		}
+	};
+
+	beforeEach(() => {
+		provider = getTestObjectProvider();
+	});
+
+	it("e2e zamboni avoids modifying segments with pending interval changes", async () => {
+		const container1 = await provider.createContainer(runtimeFactory, loaderProps);
+		const container2 = await provider.loadContainer(runtimeFactory, loaderProps);
+		const container3 = await provider.loadContainer(runtimeFactory, loaderProps);
+
+		const dataObject1 = (await container1.getEntryPoint()) as ITestFluidObject;
+		const dataObject2 = (await container2.getEntryPoint()) as ITestFluidObject;
+		const dataObject3 = (await container3.getEntryPoint()) as ITestFluidObject;
+
+		const sharedString1 = await dataObject1.getSharedObject<SharedString>(sharedType);
+		const sharedString2 = await dataObject2.getSharedObject<SharedString>(sharedType);
+		const sharedString3 = await dataObject3.getSharedObject<SharedString>(sharedType);
+
+		// C-AB
+		// D-C-AB
+		// E-HIJ-FG-D-C-AB
+		//   ^----------^
+		sharedString3.insertText(0, "AB");
```

Comment:
fwiw, I think this test may be sensitive to sequencing order of the ops in each segment, which is no longer deterministic due to the e2e nature (you can take pains to make it so, but I'm not sure).

I think we should probably lean more heavily on unit test coverage rather than e2e in this realm anyway, and I see you added some to pending state manager, so I'm not super concerned.

Reply:
Ok, I can see what I can do to fix this. You're right this can lead to test flakiness

Reply:
A PR CI hit this failure. I'll send you the result. This might need to be disabled.